helios-testing/src/main/java/com/spotify/helios/testing/Polling.java

```diff
@@ -37,13 +37,14 @@
       }
       Thread.sleep(500);
     }
-    throw new TimeoutException();
+    throw new TimeoutException(String.format(message, timeout, timeUnit.toString().toLowerCase()));
```

Comment:
Sorry I only got a chance to see this now.

Why not just use `message` instead of formatting it? Or `message` + `String.format("Timed out after %d %s", timeout, timeUnit.toString().toLowerCase())`. It's not obvious to callers that the message they pass in will be string formatted.